packages/ts-codegen/src/__tests__/index.test.ts

```diff
@@ -744,9 +744,313 @@ describe(generateTS, () => {
     });
   });
 
+  it("imports an entity, which is used to implement interface and returned by resolver", () => {
+    const { resolvers, models } = runGenerateTest(graphql`
+      extend schema @import(from: "@msteams/packages-test", defs: ["Entity"])
+      
+      interface Person implements Entity {
+          id: ID!
+      }
+
+      type User implements Person {
+        id: ID!
+      }
+
+      extend type Query {
+        userById(id: ID!): Person
+      }
+    `);
+    expect(models).toMatchInlineSnapshot(`
+      "import type { EntityModel } from "@msteams/packages-test";
+      // Base type for all models. Enables automatic resolution of abstract GraphQL types (interfaces, unions)
+      export interface BaseModel {
+          readonly __typename: string;
+      }
+      export interface PersonModel extends BaseModel, EntityModel {
+          __typename: string;
```

Comment:
I assumed the readonly was applied the same to all these `__typename` cases, but seeing this is still here I guess not.